org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -56,55 +60,84 @@ public boolean performFinish() {
-			String createCommand = "dotnet new " + template + " -o " + location.toString();
-
-			Runtime runtime = Runtime.getRuntime();
 			try {
-				Process process = runtime.exec(createCommand);
-				if (process.waitFor() != 0) {
-					MessageDialog.openError(getShell(), "Cannot create dotnet template",
-							"The 'dotnet new' command failed.");
-					return false;
-				}
-			} catch (IOException | InterruptedException e) {
+				getContainer().run(true, true, monitor -> {
+					monitor.beginTask("Creating .NET project",0);
+					String createCommand = "dotnet new " + template + " -o " + location.toString();
+					Runtime runtime = Runtime.getRuntime();
+
+					try {
+						Process process = runtime.exec(createCommand);
+						boolean isProcessDone = false;
+						while (!isProcessDone) {
+							if (monitor.isCanceled()) {
+								process.destroyForcibly();
+							}
+							isProcessDone = process.waitFor(100, TimeUnit.MILLISECONDS);
+						}
+						monitor.done();
+						if (process.exitValue() != 0) {
+							MessageDialog.openError(getShell(), "Cannot create dotnet template",
+									"The 'dotnet new' command exited with :" + process.exitValue());
+						} else {
+							createProject(projectName, location);
+						}
+					} catch (IOException | InterruptedException e) {
+						monitor.done();
+						MessageDialog.openError(getShell(), "Cannot create dotnet template",
+								"The 'dotnet new' command failed.");
+					}
+				});
+			} catch (InvocationTargetException | InterruptedException e) {
 				MessageDialog.openError(getShell(), "Cannot create dotnet template",
-						"The 'dotnet new' command failed.");
-				return false;
+						"The 'dotnet new' command failed");
 			}
+		} else {
+			createProject(projectName, location);
 		}
+		return true;
+	}
+
+	private void createProject(String name, File directory) {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		IProject project = root.getProject(name);
 
 		try {
-			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-			IProject project = root.getProject(projectName);
 			final IWorkspace workspace = ResourcesPlugin.getWorkspace();
-
 			IProjectDescription projectDescription = workspace.newProjectDescription(project.getName());
-			String projectLocation = location.getAbsolutePath();
+
+			String projectLocation = directory.getAbsolutePath();
 			IPath projectPath = new Path(projectLocation);
-			projectDescription.setLocation(projectPath);
 
+			projectDescription.setLocation(projectPath);
 			project.create(projectDescription, null);
 
-			IWorkingSetManager wsm = PlatformUI.getWorkbench().getWorkingSetManager();
-			wsm.addToWorkingSets(project, wizardPage.getWorkingSets());
+		} catch (Exception e) {
+			MessageDialog.openError(getShell(), "Unable to load project description", e.toString());
+		}
+
+		Display.getDefault().asyncExec(() -> {
+			try {
+				IWorkingSetManager wsm = PlatformUI.getWorkbench().getWorkingSetManager();
```

Comment:
`getActiveWorkbenchWindow()` and `addToWorkingSets` both must be called from the UI thread. The others will be moved.